addons/common/functions/fnc_switchPersistentLaser.sqf

```diff
@@ -0,0 +1,64 @@
+/*
+ * Author: Dystopian
+ * Controls persistent laser state.
+ *
+ * Arguments:
+ * 0: Enabled <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * true call ace_common_fnc_switchPersistentLaser
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_enabled"];
+
+if (!_enabled) exitWith {
+    ["KeyDown", GVAR(laserKeyDownEH)] call CBA_fnc_removeDisplayHandler;
+    ["weapon", GVAR(laserWeaponEH)] call CBA_fnc_removePlayerEventHandler;
+    ["turret", GVAR(laserTurretEH)] call CBA_fnc_removePlayerEventHandler;
+    ["vehicle", GVAR(laserVehicleEH)] call CBA_fnc_removePlayerEventHandler;
+};
+
+GVAR(laserKeyDownEH) = ["KeyDown", {
+    if !((_this select 1) in actionKeys "headlights") exitWith {false};
```

Comment:
Rebind `headlights` to something with modifier key, e.g. `Shift + L` and try this.

Reply:
Here we have to choose: key with modifier or persistent laser. I think this implementation is most compromise. We can add much more complicated way to handle this, but I don't think it's worth it.

Reply:
But `actionKeys` is really bad even apart from that. It reports 0 on very low FPS on keyDown, despite the key being pressed and the vanilla action executing. I just don't think this makes for reliable code.